examples/topics/stats/pyramid.py

```diff
@@ -0,0 +1,54 @@
+''' A pyramid plot showing the age distribution of the male and female Titanic passengers using the `Bokeh` hbar glyph.
+
+.. bokeh-example-metadata::
+    :sampledata: titanic
+    :apis: bokeh.plotting.figure.hbar
+    :refs: :ref:`ug_topics_stats_pyramid`
+    :keywords: pyramid, hbar
+'''
+import numpy as np
+
+from bokeh.models import CustomJSTickFormatter, Label
+from bokeh.palettes import DarkText, Vibrant3 as colors
+from bokeh.plotting import figure, show
+from bokeh.sampledata.titanic import data as df
+
+sex_group = df.groupby("sex")
+
+female_ages = sex_group.get_group("female")["age"].dropna()
+male_ages = sex_group.get_group("male")["age"].dropna()
+
+bin_width = 5
+bins = np.arange(0, 72, bin_width)
+m_hist, edges = np.histogram(male_ages, bins=bins)
+f_hist, edges = np.histogram(female_ages, bins=bins)
+
+p = figure(title="Age population pyramid of titanic passengers, by gender", height=400, width=600,
+           x_range=(-90, 90), x_axis_label="count")
+
+p.hbar(right=f_hist, y=edges[1:], height=bin_width*0.8, color=colors[0], line_width=0)
+
+p.hbar(right=m_hist * -1, y=edges[1:], height=bin_width*0.8, color=colors[1], line_width=0)
+
+# add text to every other bar
+for i, (count, age) in enumerate(zip(f_hist, edges[1:])):
+    if i % 2 == 1:
+        continue
+    p.text(x=count, y=edges[1:][i], text=[f"{age-bin_width}-{age}yrs"],
+           x_offset=5, y_offset=7, text_font_size="12px", text_color=DarkText[5])
+
+# customise x-axis and y-axis
+p.xaxis.ticker = (-60, -40, -20, 0, 20, 40, 60)
```

Comment:
```suggestion
p.xaxis.ticker = (-80, -60, -40, -20, 0, 20, 40, 60, 80)
```

Hi. Your example looks pretty great but I have a final change request. Could you please add tick for the 80 vline. In my opinion the left side looks empty.

Reply:
It could also be worth seeing what things look like without any value for `ticker` at all, i.e. the default auto-ticker might look fine. It tries to pick 8-10 tick values at "nice" locations (i.e. multiples of 2, 5, and 10)

Reply:
Here's how it looks currently:
![current](https://github.com/bokeh/bokeh/assets/99359668/af54c0aa-8f57-4a6e-9d14-74d35b0c2500)


Here's how it looks with the ticker at `(-80, 80)`:
![80](https://github.com/bokeh/bokeh/assets/99359668/19c3e5f7-e6b7-449f-8faa-46738f6a322c)

Here's it with the default ticker (with line 41 deleted):
![without](https://github.com/bokeh/bokeh/assets/99359668/48ca1224-1952-4a3f-8e38-bb8238901abc)